lychee-lib/src/checker/file.rs

```diff
@@ -105,63 +105,111 @@ impl FileChecker {
     ///
     /// Returns a `Status` indicating the result of the check.
     async fn check_path(&self, path: &Path, uri: &Uri) -> Status {
-        if path.exists() {
-            return self.check_existing_path(path, uri).await;
+        let file_path = self.resolve_file_path(path);
+        let has_fragment = uri.url.fragment().is_some_and(|x| !x.is_empty());
+
+        // If file_path exists, check this file
+        if file_path.is_some() {
+            return self.check_file(&file_path.unwrap(), uri).await;
+        }
+        // If path is a directory, and we cannot find an index file inside it,
+        // and we don't have a fragment, just return success. This is for
+        // backward compatibility.
+        else if path.is_dir() && !has_fragment {
+            return Status::Ok(StatusCode::OK);
```

Comment:
I added this logic to ensure that the following test can pass:

https://github.com/kemingy/lychee/blob/8f2f746bf932dd0d5b0de3a4364b0e01e20617c8/fixtures/fragments/file1.md?plain=1#L68-L70

While I personally prefer this test to fail, I may not be aware of all the context surrounding it.

Reply:
I just found that I forgot to add the fragment to this test case. So it doesn't test the new code in #1713.

Instead, it has been treated as a success by lychee for a long time.

I'm not sure whether it's a common case for some scenarios or not.

Reply:
@kemingy Hey 👋


If I take out the quoted code in this comment, in other words, treat `[empty](empty_dir/#no_exist_fragment)` as an error, will this cause the "Too many open files" error to reappear? I'm not entirely clear on how checking link fragments relates to the "Too many open files" error.